cmd/kubeadm/app/master/manifests.go

```diff
@@ -329,13 +329,10 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration, selfHosted bool) [
 
 	command = append(command, getAuthzParameters(cfg.AuthorizationMode)...)
 
-	// Use first address we are given
-	if len(cfg.API.AdvertiseAddresses) > 0 {
-		if selfHosted {
-			command = append(command, "--advertise-address=$(POD_IP)")
-		} else {
-			command = append(command, fmt.Sprintf("--advertise-address=%s", cfg.API.AdvertiseAddresses[0]))
-		}
+	if selfHosted {
```

Comment:
With @dgoodwin PR moving this, wouldn't it be better to check the config? This might cause a merge conflict.